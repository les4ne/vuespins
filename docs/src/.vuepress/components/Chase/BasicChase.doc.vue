<template>
  <div>
    <Chase />
  </div>
</template>

<script>
import { Chase } from 'vuespins'

export default {
  name: 'app',
  components: { Chase }
}
</script>
