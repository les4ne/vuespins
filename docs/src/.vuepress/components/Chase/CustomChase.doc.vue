<template>
  <div>
    <Chase color="#fb503b" size="80" speed="5" />
  </div>
</template>

<script>
import { Chase } from 'vuespins'

export default {
  name: 'app',
  components: { Chase }
}
</script>
