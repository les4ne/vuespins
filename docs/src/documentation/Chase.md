# Chase

## **Basic**

<br>
<client-only>
<Chase />
</client-only>

### Props

Basic "Chase" component

| property | propType | required | default | custom |
| :------: | :------: | :------: | :-----: | :----: |
| `color`  |  string  |  false   | #282c34 |   -    |
|  `size`  |  number  |  false   |   50    |   -    |
| `speed`  |  number  |  false   |   2.5   |   -    |

### Usage

<<< @/src/.vuepress/components/Chase/BasicChase.doc.vue{3,8,10-13}

## **Custom**

<br>
<client-only>
<Chase color="#fb503b" :size="80" :speed="5" />
</client-only>

### Props

Custom "Chase" component

| property | propType | required | default | custom  |
| :------: | :------: | :------: | :-----: | :-----: |
| `color`  |  string  |  false   | #282c34 | #fb503b |
|  `size`  |  number  |  false   |   50    |   80    |
| `speed`  |  number  |  false   |   2.5   |    5    |

### Usage

<<< @/src/.vuepress/components/Chase/CustomChase.doc.vue{3,8,10-13}
