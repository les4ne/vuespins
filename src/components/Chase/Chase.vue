<template>
  <div class="vs-chase" :style="cssVars">
    <div class="vs-chase-dot"></div>
    <div class="vs-chase-dot"></div>
    <div class="vs-chase-dot"></div>
    <div class="vs-chase-dot"></div>
    <div class="vs-chase-dot"></div>
    <div class="vs-chase-dot"></div>
  </div>
</template>

<script>
export default {
  name: 'vs-chase',
  props: {
    color: {
      type: String,
      default: '#282c34'
    },
    size: {
      type: String,
      default: '50'
    },
    speed: {
      type: String,
      default: '2.5'
    }
  },
  computed: {
    cssVars() {
      return {
        '--vs-chase-size': this.size + 'px',
        '--vs-chase-color': this.color,
        '--vs-chase-speed': this.speed + 's',
        '--vs-chase-delay': this.speed - 0.5 + 's',
        '--vs-chase-delay-1': this.speed - 3.6 + 's',
        '--vs-chase-delay-2': this.speed - 3.5 + 's',
        '--vs-chase-delay-3': this.speed - 3.4 + 's',
        '--vs-chase-delay-4': this.speed - 3.3 + 's',
        '--vs-chase-delay-5': this.speed - 3.2 + 's',
        '--vs-chase-delay-6': this.speed - 3.1 + 's'
      }
    }
  }
}
</script>

<style scoped>
.vs-chase {
  width: var(--vs-chase-size);
  height: var(--vs-chase-size);
  position: relative;
  animation: vs-chase var(--vs-chase-speed) infinite linear both;
  -webkit-animation: vs-chase var(--vs-chase-speed) infinite linear both;
}

.vs-chase-dot {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  animation: vs-chase-dot var(--vs-chase-delay) infinite ease-in-out both;
  -webkit-animation: vs-chase-dot var(--vs-chase-delay) infinite ease-in-out
    both;
}

.vs-chase-dot:before {
  content: '';
  display: block;
  width: 25%;
  height: 25%;
  background-color: var(--vs-chase-color);
  border-radius: 100%;
  animation: vs-chase-dot-before var(--vs-chase-delay) infinite ease-in-out both;
}

.vs-chase-dot:nth-child(1) {
  animation-delay: var(--vs-chase-delay-1);
}
.vs-chase-dot:nth-child(2) {
  animation-delay: var(--vs-chase-delay-2);
}
.vs-chase-dot:nth-child(3) {
  animation-delay: var(--vs-chase-delay-3);
}
.vs-chase-dot:nth-child(4) {
  animation-delay: var(--vs-chase-delay-4);
}
.vs-chase-dot:nth-child(5) {
  animation-delay: var(--vs-chase-delay-5);
}
.vs-chase-dot:nth-child(6) {
  animation-delay: var(--vs-chase-delay-6);
}
.vs-chase-dot:nth-child(1):before {
  animation-delay: var(--vs-chase-delay-1);
}
.vs-chase-dot:nth-child(2):before {
  animation-delay: var(--vs-chase-delay-2);
}
.vs-chase-dot:nth-child(3):before {
  animation-delay: var(--vs-chase-delay-3);
}
.vs-chase-dot:nth-child(4):before {
  animation-delay: var(--vs-chase-delay-4);
}
.vs-chase-dot:nth-child(5):before {
  animation-delay: var(--vs-chase-delay-5);
}
.vs-chase-dot:nth-child(6):before {
  animation-delay: var(--vs-chase-delay-6);
}

@keyframes vs-chase {
  100% {
    transform: rotate(360deg);
  }
}

@keyframes vs-chase-dot {
  80%,
  100% {
    transform: rotate(360deg);
  }
}

@keyframes vs-chase-dot-before {
  50% {
    transform: scale(0.4);
  }
  100%,
  0% {
    transform: scale(1);
  }
}
</style>
