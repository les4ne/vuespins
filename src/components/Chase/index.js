import Chase from './Chase.vue'

export default Chase
