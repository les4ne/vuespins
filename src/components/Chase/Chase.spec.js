import { shallowMount } from '@vue/test-utils'
import Chase from './Chase.vue'

describe('Chase.vue', () => {
  it('renders correctly', () => {
    expect(shallowMount(Chase).element).toMatchSnapshot()
  })
})
