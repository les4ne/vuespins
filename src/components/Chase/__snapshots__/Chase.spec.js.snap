// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chase.vue renders correctly 1`] = `
<div
  class="vs-chase"
  style="--vs-chase-size: 50px; --vs-chase-color: #282c34; --vs-chase-speed: 2.5s; --vs-chase-delay: 2s;"
>
  <div
    class="vs-chase-dot"
  />
   
  <div
    class="vs-chase-dot"
  />
   
  <div
    class="vs-chase-dot"
  />
   
  <div
    class="vs-chase-dot"
  />
   
  <div
    class="vs-chase-dot"
  />
   
  <div
    class="vs-chase-dot"
  />
</div>
`;
